idaplugin/rematch/collectors/identity_hash.py

```diff
@@ -0,0 +1,33 @@
+import idautils
+import idc
+
+from .vector import Vector
+
+
+class IdentityHashVector(Vector):
+  type = 'identity_hash'
+  type_version = 0
+
+  # The Keleven
+  # http://movies.stackexchange.com/q/11495
+  keleven = 17391172068829961267
+
+  def __init__(self, **kwargs):
+    super(IdentityHashVector, self).__init__(**kwargs)
+    self.hash = self.keleven
+
+  def _cycle(self, b):
+    self.hash |= 5
+    self.hash ^= b
+    self.hash *= self.hash
+    self.hash ^= (self.hash >> 32)
+    self.hash &= 0xffffffffffffffff
+
+  def _data(self):
+    for ea in idautils.FuncItems(self.offset):
+      self._cycle(idc.Byte(ea))
+      if idc.isRef(idc.GetFlags(ea)):
```

Comment:
why are you ignoring every start of bb?

this isn't
```
only ignoring reference offsets 
````

from what i understand because the following condition would also be true:

```
loc_858066BC
                MOVS            R0, #0x12
                STR             R0, [SP,#0x78+var_40]

```

you will miss the first instruction.